json_encoder.go

```diff
@@ -104,6 +105,18 @@ func (enc *jsonEncoder) AddInt64(key string, val int64) {
 // grade-school notation otherwise).
 func (enc *jsonEncoder) AddFloat64(key string, val float64) {
 	enc.addKey(key)
+	if math.IsNaN(val) {
+		enc.bytes = append(enc.bytes, `"NaN"`...)
```

Comment:
Hmm should we be encoding the same key with a different type? What do you think of the idea of using something like "key-error" or something?

It's a pain JSON doesn't allow `NaN` or infinity